### ISSUE TITLE
coding-standard.xhtml script section is misleading and outdated

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3381]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3381|
|Type|defect|
|Created|2008-08-04 14:23:01Z|
|Branch|https://github.com/twisted/twisted/tree/outdated-coding-standard-3381|

The scripts section has some problems:

  * it talks about putting things in the `bin/` directory - this needs to be a subproject bin directory now.
  * It talks about the _Twisted running-from-SVN header_ but doesn't say what it is.
  * It talks about `doc/man`, which also needs to be updated to deal with subprojects.
  * It talks about a _script list_ in setup.py, which isn't a real thing anymore (scripts are found automatically).

<details><summary>Searchable metadata</summary>

```
trac-id__3381 3381
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_outdated_coding_standard_3381 branches/outdated-coding-standard-3381
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1217859781000000 1217859781000000
changetime__1226846231000000 1226846231000000
version__None None
owner__ 
cc__exarkun cc__thijs cc__radix
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [25121]) Branching to 'outdated-coding-standard-3381'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Fixed the points in r25122. Prepared the Twisted running-from-SVN header section so that blank can be filled in when somebody tells me what this header actually looks like, since afaik I haven't seen that before. Looks like that should simply be removed though.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

You can find the running from svn header in any of the existing scripts in bin/

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Thanks for the pointer.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Added the header in r25140, putting it up for review again.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph set owner to @thijstriemstra|
|-|-|

    1. "the bin folder (of a sub-project)" is vague.  It should say something more like: "For twisted core, this file should be placed in the bin/ directory; for example, `bin/twistd`.  For sub-projects, it should be placed in `bin/<subproject>`; for example, the key-generation tool for the conch sub-project is in `bin/conch/ckeygen`."
    1. This isn't really the documentation's fault, but the running-from-SVN header is buggy.  It assumes a sub-project bin directory.  Scripts for twisted core (twistd, trial) need one fewer reference to `os.pardir`.  There should really be a separate ticket for fixing this.
    1. The running-from-SVN header and shebang lines are Python.  They should say `class="python"`, accordingly, not `"shell"`.

Personally I think the different-bin-subdirectories thing is kind of dumb.  Eventually I'd like to get away from it.  The whole point of having a bin/ directory was to allow users to download Twisted, do nothing but add one entry to `$PATH`, and then have Twisted be effectively thereby "installed".  I think the metadata about which scripts go with which sub-projects could easily be stored somewhere else.  I have refrained from making any noise about this to avoid disturbing the apparently very difficult release-automation task that has been going on, and so radix should be consulted about this before we change anything.  Depending on his opinion, we might be able to skip filing a ticket for changing the preamble to be variable as described point 2, and just go back to a fixed one-os.pardir preamble.

This doens't affect me directly in the slightest; I added a workaround to Combinator to deal with project subdirectories this a long time ago.  But, I still think it would be nicer for new users of Twisted to have this functionality.

(As a separate, related thought: if we had a parallel 'scripts' directory that had versions of all the scripts with .py extensions and a batch file to set PATHEXT, we could provide a similarly pleasant out-of-the-box experience on Windows.)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Thanks for the review, addressed points 1 and 3 in r25329. Added [#3526](https://github.com/twisted/twisted/issues/3526) for 2. Adding radix to cc for his opinion about your notes. Up for review again.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

Missed a shell/python issue, fixed in r25330.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_11"></a>|@therve set owner to @thijstriemstra|
|-|-|

Fine, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [25374]) Merge outdated-coding-standard-3381: Update for the script section of the coding standard.

Author: thijs
Reviewer: exarkun, glyph, therve
Fixes: [#3381](https://github.com/twisted/twisted/issues/3381)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
